modules/ting_new_materials/ting_new_materials.covers.inc

```diff
@@ -24,29 +37,53 @@ function ting_new_materials_check_covers($covers_ids, $number_of_objects) {
       $cover_id->hasImage = TRUE;
       $number_found++;
     } else {
-      $missing_images_local_ids[] = (object) array(
-          'localIdentifier' => $cover_id->localId,
-          'libraryCode' => $cover_id->ownerId,
-      );
+      $cover_id->hasImage = FALSE;
+      $missing_images_local_ids[] = $cover_id->localId;
     }
     if ($number_found >= $number_of_objects) {
       break;
     }
   }
-  $service = new AdditionalInformationService(variable_get('ting_covers_addi_wsdl_url'), variable_get('ting_covers_addi_username'), variable_get('ting_covers_addi_group'), variable_get('ting_covers_addi_password'), variable_get('ting_covers_addi_enforce_url', FALSE));
-  // Local ids = Faust numbers. Library object identifiers can be confusing.
-  $additional_informations = $service->getByLocalIdentifier($missing_images_local_ids);
-  foreach ($missing_images_local_ids as $local_id) {
-    if (isset($additional_informations[$local_id->localIdentifier]) && $ai = $additional_informations[$local_id->localIdentifier]) {
-      if ($ai->detailUrl || $ai->thumbnailUrl) {
-        $covers_ids[$local_id->localIdentifier]->hasImage = TRUE;
-      } else {
-        $covers_ids[$local_id->localIdentifier]->hasImage = FALSE;
+
+  try {
+    $config = _ting_cover_ddb_get_client_config();
+    $apiInstance = new CoverApi(
+      new Client(),
+      $config
+    );
+    _ting_covers_ddb_debug_log('Requested covers for pids: ' . implode(', ', $missing_images_local_ids));
```

Comment:
You should not use "private" functions in external modules.

Reply:
I known but that module is according to DDB on it's way out of core.

Reply:
If this is only logging then why keep it around?

I see why we might like logging of the pids we request covers for when the cover service is used but this promotes a bad pattern. We might as well just drop the line.

Reply:
Removed :-)